test/EndToEndTests/Services/ServiceWrapperApp/ServiceWrapperApp.cs

```diff
@@ -0,0 +1,319 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ServiceWrapperApp.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace ServiceWrapperApp
+{
+    using System;
+    using System.Linq;
+    using Microsoft.Test.OData.Framework.Server;
+    using Microsoft.Test.OData.Services.TestServices;
+    using ServiceCommand = Microsoft.Test.OData.Tests.Client.IPCCommandMap.ServiceCommands;
+    using ServiceDescriptorType = Microsoft.Test.OData.Tests.Client.IPCCommandMap.ServiceDescriptorType;
+    using ServiceType = Microsoft.Test.OData.Tests.Client.IPCCommandMap.ServiceType;
+
+    /// <summary>
+    /// Commandline tool to launch the test services for OData Client E2E tests. This tool must be run in admin mode to open ports.
+    /// </summary>
+    class ServiceWrapperApp
+    {
+        private static ServiceDescriptor serviceDescriptor;
+        private static IServiceWrapper serviceWrapper;
+        private static bool isServiceRunning = false;
+
+        /// <summary>
+        /// App entry.
+        /// </summary>
+        /// <param name="args">Arguments passed into commandline.</param>
+        static int Main(string[] args)
+        {
+            ServiceCommand serviceCommand;
+            ServiceDescriptorType serviceDescriptorType;
+            ServiceType serviceType;
+            string commandText;
+
+            if (args.Contains("h") || args.Contains("help") || args.Contains("?"))
+            {
+                PrintHelp();
+                return 0;
+            }
+
+            if (AttachDebugger(args))
+            {
+                Console.WriteLine("Attach a debugger. Press Enter to continue.");
+                commandText = Console.ReadLine();
+            }
+
+            if (!ParseArguments(args, out serviceDescriptorType, out serviceType))
+            {
+                return 1;
+            }
+
+            // Initialize the service descriptor and wrapper
+            SetServiceDescriptorAndWrapper(serviceDescriptorType, serviceType);
+
+            // Wait for a Start command to start the service
+            do

```

Comment:
Maybe add a WriteLine to explain what's happening?

Reply:
Done.